### DISCUSSION TITLE
Highlight vue components differently than plain HTML elements

### DISCUSSION THREAD

**Discussion by danrot** [nvim-treesitter/nvim-treesitter#5085] [Q&A]
I would like to highlight the tags for vue components in a `.vue` file differently than standard HTML elements, i.e. in the following example I would like the `router-view` to be highlighted in a different way than the `div` tag:

```vue
<template>
  <div>
    <router-view></router-view>
  </div>
</template>
```

However, if I use the `:Inspect` command on both tags both seem to be identified as `@tag.vue`, so there is no way for me to tell them apart. Is that something that would be possible, or is that a limitation of treesitter, and it does not work?

One way that could work is that treesitter knows about all the standard HTML tags, and apply another group to non-standard HTML tags, in which case I could assume they are vue components.

**Comment by amaanq** [comment]
It's possible but could be expensive to query, by using a not-any-of predicate to exclude all html tags
